invisible_cities/cities/components.py

```diff
@@ -114,6 +113,52 @@ def proxy(**kwds):
     return proxy
 
 
+def create_timestamp(rate: float) -> float:
+    """
+    Get rate value safely: It raises warning if rate <= 0 and
+    it sets a physical rate value in Hz.
+
+    Parameters
+    ----------
+    rate : float
+           Value of the rate in Hz.
+
+    Returns
+    -------
+    Function to calculate timestamp for the given rate with 
+    event_number as parameter.
+    """
+
+    if rate == 0:
+        warnings.warn("Zero rate is unphysical, using default "
+                      "rate = 0.5 Hz instead")
+        rate = 0.5 * units.hertz
+    elif rate < 0:
+        warnings.warn(f"Negative rate is unphysical, using "
```

Comment:
This warning outputs the rate in the default units of IC for 1/time. For example, try to see the rating for `rate = -1 * units.hertz`. It does not return the expected warning with positive 1 Hz.

Reply:
Ah, it is true. I should use `abs(rate) / units.hertz`, thanks